p2p/sentry/sentry_grpc_server.go

```diff
@@ -970,17 +975,20 @@ func (ss *GrpcServer) SendMessageToRandomPeers(ctx context.Context, req *proto_s
 func (ss *GrpcServer) SendMessageToAll(ctx context.Context, req *proto_sentry.OutboundMessageData) (*proto_sentry.SentPeers, error) {
 	reply := &proto_sentry.SentPeers{}
 
-	msgcode := eth.FromProto[ss.Protocols[0].Version][req.Id]
-	if msgcode != eth.NewBlockMsg &&
-		msgcode != eth.NewPooledTransactionHashesMsg && // to broadcast new local transactions
-		msgcode != eth.NewBlockHashesMsg {
+	msgcode, protocolVersions := ss.messageCode(req.Id)
+	if protocolVersions.Cardinality() == 0 ||
+		(msgcode != eth.NewBlockMsg &&
+			msgcode != eth.NewPooledTransactionHashesMsg && // to broadcast new local transactions
+			msgcode != eth.NewBlockHashesMsg) {
 		return reply, fmt.Errorf("sendMessageToAll not implemented for message Id: %s", req.Id)
 	}
 
 	var lastErr error
 	ss.rangePeers(func(peerInfo *PeerInfo) bool {
-		ss.writePeer("[sentry] SendMessageToAll", peerInfo, msgcode, req.Data, 0)
-		reply.Peers = append(reply.Peers, gointerfaces.ConvertHashToH512(peerInfo.ID()))
+		if protocolVersions.Contains(peerInfo.protocol) {
```

Comment:
Is an `else` not needed here?

Reply:
Because we want to send messages only to compatible peers. For instance, we want to send `MessageId_NEW_POOLED_TRANSACTION_HASHES_68` only to eth/68 peers, not to eth/67 ones.

Reply:
Ah ok, so for peers not supporting the given protocol we just don't do anything.